src/pkgmt/cli.py

```diff
@@ -95,11 +95,32 @@ def hook(uninstall, run):
     default=False,
     help="Do not ask for confirmation",
 )
-@click.option("--push/--no-push", default=True)
-@click.option("--tag/--no-tag", default=True)
+@click.option("--push/--no-push", default=None)
+@click.option("--tag/--no-tag", default=None)
 @click.option("--target", default=None)
 def version(yes, push, tag, target):
     """Create a new package version"""
+
+    cfg_tag, cfg_push = config.PyprojectConfig().get_version_configurations()
```

Comment:
you could write this CLI validation logic inside PyprojectConfig and could generalize it so you only write it once and pass all arguments through it, example:

```
# cfg already contains the resolved values, and internally it prints any messages to the user if there are conflicts
cfg = config.PyprojectConfig().resolve_arguments(tag=tag, push=push)
```

Reply:
moved this to Config